src/modules/auth/native.auth.guard.ts

```diff
@@ -57,6 +57,7 @@ export class NativeAuthGuard implements CanActivate {
                         'X-Native-Auth-Address',
                         this.impersonateAddress,
                     );
+                    req.auth.address = this.impersonateAddress;
```

Comment:
It will not work it here. Might need an else for this if statement with the `req.auth = userInfo` in it

Reply:
Done